freqtrade/freqtradebot.py

```diff
@@ -443,33 +442,28 @@ def process_maybe_execute_buy(self) -> None:
         except DependencyException as exception:
             logger.warning('Unable to create trade: %s', exception)
 
-    def process_maybe_execute_sell(self, trade: Trade) -> bool:
+    def process_maybe_execute_sells(self, trades: List[Any]) -> None:
         """
-        Tries to execute a sell trade
-        :return: True if executed
+        Tries to execute sell trades in a safe way
         """
-        try:
-            self.update_trade_state(trade)
-
-            if self.strategy.order_types.get('stoploss_on_exchange') and trade.is_open:
-                result = self.handle_stoploss_on_exchange(trade)
-                if result:
-                    self.wallets.update()
-                    return result
-
-            if trade.is_open and trade.open_order_id is None:
-                # Check if we can sell our current pair
-                result = self.handle_trade(trade)
-
-                # Updating wallets if any trade occured
-                if result:
-                    self.wallets.update()
-
-                return result
-
-        except DependencyException as exception:
-            logger.warning('Unable to sell trade: %s', exception)
-        return False
+        for trade in trades:
+            try:
+                self.update_trade_state(trade)
+
+                if trade.is_open:
+                    result = False
```

Comment:
Can you do the following? 

* move `result = False` out of the loop
* run `wallets.update()` after the loop, based if any trade was changed, so it runs only once per "sell"-iteration if necessary, even if more than one sell occured.

Reply:
It's even better! 👍 

* fixed the logic in `process_maybe_execute_sells()`
* I removed the `if trade.is_open` check there -- it's redundant, the `trades` list of trades prepared in `process()` is already a list of open trades. Even if it's updated somewhere in the `Trades.update()` (seems it's not), the `handle_trade()` will simply quit with the exception
* The exception in the `handle_trade()` changed from ValueError to DependencyException to be catched safely here in process_maybe_execute_sells()